controller_interface/include/semantic_components/force_torque_sensor.hpp

```diff
@@ -81,56 +80,54 @@ class ForceTorqueSensor : public SemanticComponentInterface<geometry_msgs::msg::
     check_and_add_interface(interface_torque_x, 3);
     check_and_add_interface(interface_torque_y, 4);
     check_and_add_interface(interface_torque_z, 5);
-
-    // Set default force and torque values to NaN
-    std::fill(forces_.begin(), forces_.end(), std::numeric_limits<double>::quiet_NaN());
-    std::fill(torques_.begin(), torques_.end(), std::numeric_limits<double>::quiet_NaN());
   }
 
-  virtual ~ForceTorqueSensor() = default;
-
   /// Return forces.
   /**
    * Return forces of a FTS.
    *
-   * \return array of size 3 with force values.
+   * \return array of size 3 with force values (x, y, z).
    */
   std::array<double, 3> get_forces()
   {
-    size_t interface_counter = 0;
-    for (size_t i = 0; i < 3; ++i)
+    std::array<double, 3> forces{
+      {double_limits::quiet_NaN(), double_limits::quiet_NaN(), double_limits::quiet_NaN()}};
+    size_t interface_counter{0};
+    for (size_t i{0}; i < forces.size(); ++i)
     {
       if (existing_axes_[i])
       {
-        forces_[i] = state_interfaces_[interface_counter].get().get_value();
+        forces[i] = state_interfaces_[interface_counter].get().get_value();
         ++interface_counter;
       }
     }
-    return forces_;
+    return forces;
   }
 
   /// Return torque.
   /**
    * Return torques of a FTS.
    *
-   * \return array of size 3 with torque values.
+   * \return array of size 3 with torque values (x, y, z).
    */
   std::array<double, 3> get_torques()
   {
+    std::array<double, 3> torques{
+      {double_limits::quiet_NaN(), double_limits::quiet_NaN(), double_limits::quiet_NaN()}};
     // find out how many force interfaces are being used
     // torque interfaces will be found from the next index onward
-    auto torque_interface_counter =
-      static_cast<size_t>(std::count(existing_axes_.begin(), existing_axes_.begin() + 3, true));
+    auto torque_interface_counter{static_cast<size_t>(
+      std::count(existing_axes_.begin(), existing_axes_.begin() + forces_size_, true))};
```

Comment:
```suggestion
    const auto torque_interface_counter = static_cast<size_t>(
      std::count(existing_axes_.begin(), existing_axes_.begin() + forces_size_, true));
```
Yes, technically this is more efficient if the compiler didn't know how to optimize our code.

Let's trust it though that it can figure out how to convert this simple const assignment to something efficient while we retain the universal power of readability with `=`

Reply:
I haven't add const since this counter is incremented in the loop.